Document font-palette & @font-palette-values with its descriptors (#22519)
mdn/content: files/en-us/web/css/@font-palette-values/override-colors/index.md

---
title: override-colors
slug: Web/CSS/@font-palette-values/override-colors
page-type: css-at-rule-descriptor
tags:
  - CSS
  - CSS Fonts
  - Color Fonts
  - CSS Descriptor
  - Reference
browser-compat: css.at-rules.font-palette-values.override-colors
---

{{CSSRef}}

> The [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) **`override-colors`** is used to override colors in the chosen [base-palette](/en-US/docs/Web/CSS/@font-palette-values/base-palette) for a color font.

Reviewer: ```suggestion
The **`override-colors`** CSS [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) is used to override colors in the chosen [base-palette](/en-US/docs/Web/CSS/@font-palette-values/base-palette) for a color font.
```

## Syntax

```css
/* basic syntax */

> override-colors: <index of color> <color>;

Reviewer: ```suggestion
override-colors: <index of palette in the font> <new color>;
```

Author: This is not correct.
There are 2 separate indexes involved, the index of the base-palette used and the index of the colors in the palette.
So a font may have:
- 4 base-palettes 0-3
- 5 colors in each base-palette 0-4
override color changes the color at a specific index rather than changing the palette

/* using colour names */
override-colors: 0 red;
/* using hex-color */
override-colors: 0 #f00;
/* using rbg */
override-colors: 0 rgb(255, 0, 0);
/* overriding multiple colors */
override-colors: 0 #f00, 1 #0f0, 2 #00f;
/* overriding multiple colors with readability */
override-colors:
  0 #f00,
  1 #0f0,
  2 #00f;

> ```

Reviewer: just add a line spacing before every comment

Author: will do

Author: This is now done shall update the PR

> The `override-colors` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) takes a comma-separated list of the color index and new color value.

Reviewer: ```suggestion
The `override-colors` takes a comma-separated list of the palette index and the new color value.
```

Author: See previous comment about override-colors
https://github.com/mdn/content/pull/22519#discussion_r1039496038

Author: The `override-colors` takes a comma-separated list of the color index and the new color value.

Author: This is now done shall update the PR

The color index is zero-based and any [color value](/en-US/docs/Web/CSS/color_value) can be used.

> For each key/value pair of index/color the color with the index in the specified [base-palette](/en-US/docs/Web/CSS/@font-palette-values/base-palette) will be overwritten. If the color font does not have a color at the specified index it will be ignored.

Reviewer: ```suggestion
For each key-value pair of index and color, the color with the index in the specified [base-palette](/en-US/docs/Web/CSS/@font-palette-values/base-palette) will be overwritten. If the color font does not have a palette at the specified index, it will be ignored.
```

Author: See previous comment about override-colors
https://github.com/mdn/content/pull/22519#discussion_r1039496038

Author: For each key-value pair of index and color, the color with the index in the specified [base-palette](/en-US/docs/Web/CSS/@font-palette-values/base-palette) will be overwritten. If the color font does not have a color at the specified index, it will be ignored.

### Values

- `[ <integer [0,∞]> <absolute-color-base> ]`
  - : Specifies the index of a color in a [base-palette](/en-US/docs/Web/CSS/@font-palette-values/base-palette) and the color to overwrite it with.

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

### Changing colors of emojis

> The [Noto Color Emoji](https://fonts.google.com/noto/specimen/Noto+Color+Emoji/) font, is a color font which you can override the colors to match the sites brand.

Reviewer: ```suggestion
This example shows how to override colors in the [Noto Color Emoji](https://fonts.google.com/noto/specimen/Noto+Color+Emoji/) color font to match your site's brand.
```

#### HTML

```html
<section class="hats">
  <div class="hat">
    <h2>Original Hat</h2>
    <div class="emoji">🎩</div>
  </div>
  <div class="hat">
    <h2>Red Hat</h2>
    <div class="emoji red-hat">🎩</div>
  </div>
</section>
```

#### CSS

```css hidden
.hats {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}
```

```css
@font-face {
  font-family: "Noto Color Emoji";
  font-style: normal;
  font-weight: 400;
  src: url(https://fonts.gstatic.com/l/font?kit=Yq6P-KqIXTD0t4D9z1ESnKM3-HpFabts6diywYkdG3gjD0U&skey=a373f7129eaba270&v=v24) format("woff2");

> }
> .emoji {

Reviewer: ```suggestion
}

.emoji {
```
Similarly, each rule block will need a line space before it

  font-family: "Noto Color Emoji";
  font-size: 3rem;
}
@font-palette-values --red {
  font-family: "Noto Color Emoji";
  override-colors: 
    0 rgb(74, 11, 0), 
    1 rgb(149, 22, 1), 
    2 rgb(183, 27, 1), 
    3 rgb(193, 28, 1), 
    4 rgb(230, 34, 1);
}
.red-hat {
  font-palette: --red;
}
```

#### Result

{{EmbedLiveSample("Changing colors of emojis")}}

### Changing a color in an alternate base-palette

> As an example, using [Rocher Color Font](https://www.harbortype.com/fonts/rocher-color/) one color in the font is overridden.

Reviewer: ```suggestion
Using [Rocher Color Font](https://www.harbortype.com/fonts/rocher-color/), this example shows how to override one color in the font.
```

#### HTML

```html
<h2 class="normal-palette">Normal Palette</h2>
<h2 class="override-palette">Override Palette</h2>
```

#### CSS

```css
@font-face {
  font-family: 'Rocher';
  src: url('[path-to-font]/RocherColorGX.woff2') format('woff2'); 
}
h2 {
  font-family: 'Rocher';
}
@font-palette-values --override-palette {
  font-family: 'Rocher';
  base-palette: 3;
}
@font-palette-values --override-palette {
  font-family: 'Rocher';
  base-palette: 3;
  override-colors: 0 rebeccapurple;
}
.normal-palette {
  font-palette: --normal-palette;
}
.override-palette {
  font-palette: --override-palette;
}
```

#### Result

> Example showing the color at `index` 0 being overriden in `base-palette: 3;` with `rebeccapurple`

Reviewer: ```suggestion
This example shows the that in `base-palette` `3`, the color at index 0 is overridden with `rebeccapurple`.
```
